### PR TITLE
deprecate scp_if_ssh

### PR DESCRIPTION
**Pull request by bcoca**
ssh_transfer_method has overridden this for a long time already

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

connection/ssh

### CODE REVIEW

File: lib/ansible/plugins/connection/ssh.py [hunk 1]
```diff
@@ -289,12 +289,17 @@
...
         env: [{name: ANSIBLE_SCP_IF_SSH}]
         ini:
         - {key: scp_if_ssh, section: ssh_connection}
         vars:
           - name: ansible_scp_if_ssh
             version_added: '2.7'
+        deprecated:
+              why: In favor of the "ssh_transfer_method" option.
+              version: "2.16"
+              alternatives: ssh_trasnfer_method
```
Comment (sivel) [inline lib/ansible/plugins/connection/ssh.py]:
```suggestion
              alternatives: ssh_transfer_method
```

The only thing I would say here is that the `ansible_ssh_transfer_method` var was just added for 2.12, so it isn't a full replacement until 2.12 when the deprecation begins.

**Comment by bcoca** [comment]
then i'll split this, put this PR in for 2.13 and create docs pr with just the note about the override

**Review by sivel — APPROVED** [review]

**Pull request merged**